internal/client/job.go

```diff
@@ -50,9 +50,18 @@ func (c *Project) job() *pb.Job {
 
 // doJob will queue and execute the job. If the client is configured for
 // local mode, this will start and target the proper runner.
-func (c *Project) doJob(ctx context.Context, job *pb.Job, ui terminal.UI) (*pb.Job_Result, error) {
+// the mon channel can be used to monitor the job status as it changes.
+// The receiver must be careful to not block sending to mon as it will block
+// the job state processing loop.
+func (c *Project) doJob(ctx context.Context, job *pb.Job, ui terminal.UI, mon chan pb.Job_State) (*pb.Job_Result, error) {
 	log := c.logger
 
+	// Be sure that the monitor is closed so the reciever knows for sure the job isn't going
+	// anymore.
+	if mon != nil {
```

Comment:
Also, nitpick but I like it when channels end in `Ch` so I know they're channels. Its a LITTLE weird but seems like fairly idiomatic too?